exotations/task_maps/task_map/src/JointLimit.cpp

```diff
@@ -59,14 +59,18 @@ void JointLimit::Initialize()
 {
     double percent = init_.SafePercentage;
 
-    std::vector<std::string> jnts = scene_->getJointNames();
-    N = jnts.size();
+    N = scene_->getSolver().getNumControlledJoints();
+
+    // TODO: Strictly speaking this is incorrect as joint limits can be changed during runtime.
     Eigen::MatrixXd limits = scene_->getSolver().getJointLimits();
-    low_limits_ = limits.col(0);
```

Comment:
I am unclear why this now would result in a runtime segmentation fault independent of whether ``getJointLimits()`` returns an ``Eigen::MatrixXd`` (copy), a ``const Eigen::MatrixXd&``, or a ``Eigen::MatrixXd&``.

Reply:
Let's keep an eye on these Eigen issues.
It looks good otherwise.